react/features/base/conference/actions.js

```diff
@@ -146,6 +154,27 @@ function _addConferenceListeners(conference, dispatch, state) {
             }
         });
 
+    conference.on(
+        JitsiConferenceEvents.AUDIO_UNMUTE_PERMISSIONS_CHANGED,
+        disableAudioMuteChange => {
+            const muted = isAudioMuted(state);
+
+            // Disable the mute button only if its muted.
+            if (!disableAudioMuteChange || (disableAudioMuteChange && muted)) {
```

Comment:
I thought this complicated if would not be necessary, isn't it possible to store this state and then in the button logic, make it disabled only when muted and there's no permission to unmute?

Reply:
a logic like this is easily broken by a timing of events, whether a user was muted or unmuted at the time of the arrival, but I don't really know how this feature was designed end to end so may be missing something